openquake/hazardlib/geo/surface/simple_fault.py

```diff
@@ -233,6 +234,43 @@ def get_fault_patch_vertices(cls, fault_trace, upper_seismogenic_depth,
         return p0, p1, p2, p3
 
     @classmethod
+    def hypocentre_patch_index(cls, hypocentre, fault_trace,
+                               upper_seismogenic_depth,
+                               lower_seismogenic_depth, dip):
+        """
+        Indicate the fault patch where the hypocentre locates
+
+        :param hypocentre:
+            :class:`~openquake.hazardlib.geo.point.Point` object
+            representing the location of hypocentre.
+        :param openquake.hazardlib.geo.line.Line fault_trace:
+            Geographical line representing the intersection between
+            the fault surface and the earth surface.
+        :param upper_seismo_depth:
+            Minimum depth ruptures can reach, in km (i.e. depth
+            to fault's top edge).
+        :param lower_seismo_depth:
+            Maximum depth ruptures can reach, in km (i.e. depth
+            to fault's bottom edge).
+        :param dip:
+            Dip angle (i.e. angle between fault surface
+            and earth surface), in degrees.
+        :return:
```

Comment:
An integer corresponding to the index of the fault patch which contains the hypocentre